docassemble/ALWeaver/interview_generator.py

```diff
@@ -1558,14 +1572,26 @@ def _guess_role(self, title: str):
 
         return "unknown"
 
+    def _null_group_fields(self):
+        return {"Screen 1": [field.variable for field in self.all_fields.custom()]}
+
     def auto_group_fields(self):
         """
         Use FormFyxer to assign fields to screens.
         To assist with "I'm feeling lucky" button
         """
-        field_grouping = formfyxer.cluster_screens(
-            [field.variable for field in self.all_fields.custom()]
-        )
+        try:
+            field_grouping = formfyxer.cluster_screens(
+                [field.variable for field in self.all_fields.custom()],
+                tools_token=get_config("assembly line", {}).get(
+                    "tools.suffolklitlab.org api key", None
+                ),
+            )
+        except:
+            log(
+                f"Auto field grouping failed. Tried using tools.suffolklitlab.org api key {get_config('assembly line',{}).get('tools.suffolklitlab.org api key', None)}"
+            )
```

Comment:
Minor, but this should probably be shown to the user too, like `log(..., 'warning')`. Idk if that also shows in the logs, could just do it twice though